pageserver/src/pgdatadir_mapping.rs

```diff
@@ -854,6 +854,49 @@ impl Timeline {
     }
 }
 
+pub const AUX_FILES_V2: bool = true;
+
+fn split_prefix<'a, 'b>(path: &'a str, prefix: &'b str) -> Option<&'a str> {
+    if path.starts_with(prefix) {
+        Some(&path[prefix.len()..])
+    } else {
+        None
+    }
+}
+
+fn hash_to_raw_key(prefix: u16, data: &[u8]) -> [u8; RAW_KEY_SIZE] {
+    let mut key = [0; RAW_KEY_SIZE];
+    let hash = crc32c::crc32c(data).to_be_bytes();
+    let prefix = prefix.to_be_bytes();
+    key[0] = prefix[0];
+    key[1] = prefix[1];
+    key[2] = hash[0];
+    key[3] = hash[1];
+    key[4] = hash[2];
+    key[5] = hash[3];
+    key
+}
+
+pub fn encode_aux_file_key(path: &str) -> anyhow::Result<Key> {
```

Comment:
+1, like this approach. Detect prefixes and allow new prefixes with an error message

Reply:
Alternative I proposed: use two hashes, one for the `dirname(path)` and other one for `basename(path)`. With the assumption of having fewer dirnames, we should not need many bits for that (like here: u16).